### PR TITLE
Fix Nemotron-H gated RMSNorm fallback

### PR DESCRIPTION
**Pull request by vklimkov-nvidia**
## Summary

- match upstream Mamba gate-before-normalization semantics in the PyTorch fallback
- normalize independently in contiguous RMS groups and restore the input dtype after applying weights
- add a focused CPU regression test for grouped gated RMSNorm behavior

## Testing

- `/home/vklimkov/miniconda3/envs/nemotts/bin/pytest -q tests/collections/tts/test_nemotron_h_decoder.py` (22 passed)
- `conda run -n nemotts python setup.py style --scope nemo/collections/tts/modules/nemotron_h_decoder.py`
- `conda run -n nemotts python setup.py style --scope tests/collections/tts/test_nemotron_h_decoder.py`

### CODE REVIEW

File: nemo/collections/tts/modules/nemotron_h_decoder.py [hunk 1]
```diff
@@ -407,15 +407,17 @@ def forward(self, hidden_states: torch.Tensor, gate: Optional[torch.Tensor] = No
                 norm_before_gate=False,
             )
         else:
```
Comment (blisc) [inline nemo/collections/tts/modules/nemotron_h_decoder.py:409]: The last time I checked this path, it was completely broken. With this fix, will TTS inference results now match if use_triton is false? Maybe we should just remove the fallbacks.

File: tests/collections/tts/test_nemotron_h_decoder.py [hunk 2]
```diff
@@ -50,6 +51,24 @@ def fixture(func):
 )
 
 
+def test_mamba_rmsnorm_gated_pytorch_fallback_matches_grouped_reference():
```
Comment (blisc) [inline tests/collections/tts/test_nemotron_h_decoder.py:55]: I feel like this test should test with use_triton true and with it false, and checks torch.isclose instead of dtype matching.

File: nemo/collections/tts/modules/nemotron_h_decoder.py (hunk 1 above)
Comment (vklimkov-nvidia) [reply nemo/collections/tts/modules/nemotron_h_decoder.py:409]: yeah, i stumbled upon completely unintelligible speech in new environment without triton kernels. the fix makes it work. the speed gains keeps triton preferred ofc. removing fallback is definitely an option, although with custom kernels it's nice to have vanilla implementation for testing. what is general approach in Speech repo? cc @pzelasko

File: tests/collections/tts/test_nemotron_h_decoder.py (hunk 2 above)
Comment (vklimkov-nvidia) [reply tests/collections/tts/test_nemotron_h_decoder.py:55]: Good point on covering parity between the two paths. Addressed in f99d62a16: I kept the CPU fallback-vs-reference test so it runs without optional compiled dependencies, and added a CUDA-gated test that compares the real Triton result with a forced PyTorch fallback. The existing test was already comparing values with `torch.testing.assert_close` (which checks dtype by default), so I removed the redundant standalone dtype assertion. The new parity test passes in an environment with `mamba-ssm`. This verifies parity at the RMSNorm operator level; the no-Triton TTS smoke test produced intelligible speech, but I am not claiming exact end-to-end numerical parity.

File: tests/collections/tts/test_nemotron_h_decoder.py [hunk 3]
```diff
@@ -39,8 +39,10 @@
 
 import torch
 
+import nemo.collections.tts.modules.nemotron_h_decoder as nemotron_h_decoder
```
[the remark this answers is not in the document]
Comment (blisc) [reply tests/collections/tts/test_nemotron_h_decoder.py]: Let's fix this and test. And we can merge

**Review by blisc — DISMISSED** [review]

**Review by blisc — APPROVED** [review]
